projects/dev/src/app/group/group-manage/pending-request/pending-request.component.ts

```diff
@@ -1,108 +1,190 @@
+    if (succ.false === undefined || succ.false === 0) {
+      this.messageService.add({
+        severity: "success",
+        summary: "Success",
+        detail: `${succ.true} request(s) have been ${msg}`,
+        life: TOAST_LENGTH,
+      });
+    } else if (succ.true === undefined || succ.true === 0) {
+      this.messageService.add({
+        severity: "error",
+        summary: "Error",
+        detail: `Unable to ${verb} the selected request(s).`,
+        life: TOAST_LENGTH,
+      });
+    } else {
+      this.messageService.add({
+        severity: "warn",
+        summary: "Partial success",
+        detail: `${succ.true} request(s) have been ${msg}, ${succ.false} could not be executed`,
+        life: TOAST_LENGTH,
       });
     }
   }
 
-  constructor(private groupService: GroupService) {}
+  _processRequestError(_err) {
+    this.messageService.add({
+      severity: "error",
+      summary: "Error",
+      detail: ERROR_MESSAGE.fail,
+      life: TOAST_LENGTH,
+    });
+  }
+
+  isAccepting() {
+    return this.onGoingActivity === Activity.Accepting;
+  }
+
+  isRejecting() {
+    return this.onGoingActivity === Activity.Rejecting;
+  }
+
+  isIdle() {
+    return this.onGoingActivity === Activity.None;
+  }
+
+  constructor(
+    private groupService: GroupService,
+    private messageService: MessageService
+  ) {}
 
   ngOnInit() {
     this.panel.push({
       name: "Pending Requests",
       columns: this.columns,
     });
-    this.groupService
-      .getManagedRequests(this.id)
-      .subscribe((reqs: PendingRequest[]) => {
-        this._setRequestData(reqs);
-      });
   }
 
-  ngOnChanges(changes: SimpleChanges) {
-    this.groupService
-      .getManagedRequests(this.id)
-      .subscribe((reqs: PendingRequest[]) => {
-        this._setRequestData(reqs);
-      });
+  ngOnChanges(_changes: SimpleChanges) {
+    this.selection = [];
+    this._reloadData();
   }
 
-  onExpandWidth(e) {}
+  onAcceptOrReject(action: Action) {
+    if (this.selection.length === 0 || this.onGoingActivity !== Activity.None) {
+      return;
+    }
+
+    let resultObserver: Observable<RequestActionResponse>;
+    this.onGoingActivity = (action === Action.Accept) ? Activity.Accepting : Activity.Rejecting;
+    
+    const group_ids = this.selection.map((req: PendingRequest) => req.joining_user.group_id);
+
+    if (action === Action.Accept) {
+      resultObserver = this.groupService.acceptJoinRequest(this.id, group_ids);
+    } else {
+      resultObserver = this.groupService.rejectJoinRequest(this.id, group_ids);
+    }
 
-  onClickAccept(e) {
-    this.onAccept.emit(e);
+    resultObserver
+    .pipe(
+      tap((res: RequestActionResponse) => {
+        if (res.success === false || res.message !== "updated" || typeof res.data !== "object") {
+          throw "Unknown error";
+        }
+      })
+    )
+    .subscribe(
+      (res: RequestActionResponse) => {
+        this._displayResponseToast(
+          res,
+          action,
```

Comment:
You pass the enum as a string. Do a proper conversion like the one of the next line. One should be able to change the enum values without risking to create a bug of the UI.

Reply:
fixed.